core/src/main/java/org/apache/calcite/runtime/Hook.java

```diff
@@ -159,15 +159,6 @@ public <T> Closeable addThread(final Consumer<T> handler) {
     return () -> removeThread(handler);
   }
 
-  // CHECKSTYLE: IGNORE 1
-  /** @deprecated Use {@link #addThread(Consumer)}. */
-  @SuppressWarnings("Guava")
-  @Deprecated // to be removed before 2.0
-  public <T, R> Closeable addThread(
-      final com.google.common.base.Function<T, R> handler) {
-    return addThread((Consumer<T>) handler::apply);
-  }
```

Comment:
With guava 32 it starts failing on errorprone checks with `CheckReturnValue`
Since it is already deprecated with  a comment what to use instead probably it makes sense to remove it